http/server.go

```diff
@@ -141,6 +155,7 @@ RESET:
 
 		select {
 		case <-ctx.Done():
```

Comment:
Could we move this `ctx.Done()` to the new `select` at https://github.com/drand/drand/pull/843/files#diff-9cdb0ee2b48996738ec1fc3037630e10eb388ebc30fb400df2e627590f0148c6R122?